packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -1833,11 +1859,29 @@ function resetChildLanes(completedWork: Fiber) {
         newChildLanes,
         mergeLanes(child.lanes, child.childLanes),
       );
+
+      if (!didBailout) {
```

Comment:
Nit: It might be better to fork this loop into two and check the boolean before entering each loop instead of checking it for each iteration.

In fact, we probably could just copy the childLanes field from "current" since it must be the same then.

Reply:
It seems like just copying the childLanes doesn't seem to work. Just doing the former for now. @bvaughn can maybe look into it more tomorrow?

Reply:
@lunaruan You can only copy from `childLanes` when `didBailout` is true. Is that what you tried? If that fails some tests I'd be curious to know which ones.

Reply:
In fact, you shouldn't even have to copy it in the `didBailout` path because it should already equal the one on `current`. Might be some edge case where this doesn't hold true, but I can't think of one and if any exists it's probably wrong.

Reply:
Yeah when `didBailout` is `true` I tried setting `completedWork.childLanes` to `completedWork.alternate.childLanes`. I didn't investigate much yet but a ton of tests failed.

Reply:
I find these multi-line forks get a little hard to trace through, but you're right we avoid re-checking `didBailout` a bunch this way so... :smile: fair enough!